boards/wio_terminal/src/wifi/wifi_rpcs.rs

```diff
@@ -0,0 +1,396 @@
+#[allow(dead_code)]
+use super::erpc::{codec, id, Err};
+use generic_array::{ArrayLength, GenericArray};
+use heapless::{consts::U18, String};
+use nom::{
+    lib::std::ops::RangeFrom, lib::std::ops::RangeTo, number::streaming, InputIter, InputLength,
+    Slice,
+};
+
+pub struct GetMacAddress {}
+
+impl codec::RPC for GetMacAddress {
+    type ReturnValue = String<U18>;
+    type Error = i32;
+
+    fn header(&self, seq: u32) -> codec::Header {
+        codec::Header {
+            sequence: seq,
+            msg_type: id::MsgType::Invocation,
+            service: id::Service::Wifi,
+            request: id::WifiRequest::GetMacAddress.into(),
+        }
+    }
+
+    fn parse(&mut self, data: &[u8]) -> Result<Self::ReturnValue, Err<Self::Error>> {
+        let (data, hdr) = codec::Header::parse(data)?;
+        if hdr.msg_type != id::MsgType::Reply
+            || hdr.service != id::Service::Wifi
+            || hdr.request != id::WifiRequest::GetMacAddress.into()
+        {
+            return Err(Err::NotOurs);
+        }
+
+        if data.input_len() < 18 {
+            return Err(Err::RPCErr(-1));
+        }
+        let mut mac: String<U18> = String::new();
+        for b in data.slice(RangeTo { end: 17 }).iter_elements() {
+            mac.push(b as char).map_err(|_| Err::ResponseOverrun)?;
+        }
+
+        let (_, result) = streaming::le_u32(data.slice(RangeFrom { start: 18 }))?;
+        if result != 0 {
+            Err(Err::RPCErr(result as i32))
+        } else {
+            Ok(mac)
+        }
+    }
+}
+
+pub struct IsScanning {}
+
+impl codec::RPC for IsScanning {
+    type ReturnValue = bool;
+    type Error = ();
+
+    fn header(&self, seq: u32) -> codec::Header {
+        codec::Header {
+            sequence: seq,
+            msg_type: id::MsgType::Invocation,
+            service: id::Service::Wifi,
+            request: id::WifiRequest::IsScanning.into(),
+        }
+    }
+
+    fn parse(&mut self, data: &[u8]) -> Result<Self::ReturnValue, Err<Self::Error>> {
+        let (data, hdr) = codec::Header::parse(data)?;
+        if hdr.msg_type != id::MsgType::Reply
+            || hdr.service != id::Service::Wifi
+            || hdr.request != id::WifiRequest::IsScanning.into()
+        {
+            return Err(Err::NotOurs);
+        }
+
+        if data.input_len() < 1 {
+            return Err(Err::RPCErr(()));
+        }
+        Ok(data.iter_elements().nth(0) != Some(0))
+    }
+}
+
+#[derive(Debug, Copy, Clone)]
+#[repr(u32)]
+pub enum BssType {
+    Infra = 0,
+    Adhoc = 1,
+    Any = 2,
+    Unknown = core::u32::MAX,
+}
+
+const WEP_ENABLED: u32 = 1;
```

Comment:
You could use https://crates.io/crates/bitflags for this

Reply:
Done (but in different crate now)